### DISCUSSION TITLE
Installation also starts if there is no internet connection

### DISCUSSION THREAD

**Discussion by revast** [Security-Onion-Solutions/securityonion#12376] [2.4]
### Version

2.4.40

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

4

### RAM

32

### Storage for /

300

### Storage for /nsm

600

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

When entering a gateway in Setup, it seems that connection to internet is not checked ;
- One could enter the wrong one, click and write 20 minutes through the setup process, only to find that installation will fail. 
   - It would be nice to have a means to being prompted that there was a "problem connecting to SO-servers" and one has a means to enter the gateway address again.

- It could also be that the gateway was entered correctly, and some firewall setting is blocking internet access.
   - Would be also nice if setup could inform that it "found the gateway, but still can't connect to internet."  
- Also when Setup is finished, and Installation begins, the installation again does not check for a working internet connection, and thus fails catastrophically.
   - It would be nice to also have a prompt there which tells what's going on....
   
- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by reyesj2** [comment]
If you're having trouble with internet you can always just install SO in airgap mode and that will only use resources available on the ISO and not depend on any external network resources. https://docs.securityonion.net/en/2.4/airgap.html#airgap
